docs(hcp): add HCP organization constraints documentation (#2535)
hashicorp/web-unified-docs: content/hcp-docs/content/docs/hcp/admin/enterprise-constraints.mdx

- HCP Vault
- HCP Boundary
- HCP Vault Radar
- HCP Packer
- HCP Waypoint

~> **Note:** HCP Terraform is not affected by enterprise constraints.

### Considerations

- **Organization-level only**: Constraints can only be applied at the organization level. You cannot create project-specific overrides or exceptions.
- **Product-level restrictions**: Constraints currently restrict entire products. You cannot restrict specific features.

## Prerequisites

To configure enterprise constraints, you must have one of the following roles:

- Organization Owner
- Organization Admin

You should also understand which HCP products your teams currently use or plan to use.

## Enable enterprise constraints

To enable enterprise constraints:

1. At the top, click the dropdown to open the organization and project selector. Select **View all organizations**.
1. Click the name of the organization.
1. From the **Organization dashboard**, click **Organization settings**.
1. Click **Enterprise constraints** in the left navigation.
1. Select the products you want to restrict from the available list:
   - HCP Vault
   - HCP Boundary
   - HCP Vault Radar
   - HCP Packer

>    - HCP Waypoint

Reviewer: Is vagrant missing?

Author: actually missing in two places. I added it 👍

---
page_title: Enterprise constraints
description: |-
  Restrict which HCP products can be deployed in your organization for centralized governance and cost control.
# START AUTO GENERATED METADATA, DO NOT EDIT
created_at: 2026-05-29T22:06:00-07:00
last_modified: 2026-05-29T22:06:00-07:00
# END AUTO GENERATED METADATA
---

# Enterprise constraints

> This page describes how to configure and manage enterprise constraints in HashiCorp Cloud Platform (HCP). Enterprise constraints allow you to restrict which HCP products can be deployed within your organization, providing centralized governance and cost control.

Reviewer 2: ```suggestion
This page describes how to configure and manage enterprise constraints in HashiCorp Cloud Platform (HCP). Enterprise constraints allow you to restrict which HCP products your organization can deploy, providing centralized governance and cost control.
```
Style guide nit: use active voice

@include 'beta-feature.mdx'

## Introduction

> An _enterprise constraint_ is a preventative control that restricts which HCP products can be deployed within your organization. Enterprise constraints work alongside Role-Based Access Control (RBAC) to provide an additional layer of governance. While RBAC determines who can perform actions, enterprise constraints determine what types of resources can be created in your organization, regardless of individual user permissions.

Reviewer 2: ```suggestion
An **enterprise constraint** is a preventative control that restricts which HCP products your organization may deploy. Enterprise constraints work alongside Role-Based Access Control (RBAC) to provide an additional layer of governance. While RBAC determines who can perform actions, enterprise constraints determine what types of resources your organization may create, regardless of individual user permissions.
```
Style nits: 
- Use active voice
- [Introduce new terms with bold](https://github.com/hashicorp/web-unified-docs/blob/main/docs/style-guide/general/fonts-and-formats.md#use-boldface-to-introduce-new-terms). We previously used italics, but we changed that last year with the combined style guide.

> Enterprise constraints apply at the organization level and automatically inherit to all projects within the organization. When you enable a constraint for a product, users cannot create new resources for that product, even if their role would otherwise allow it. Existing resources continue to operate normally when you enable a constraint.

Reviewer 2: ```suggestion
Enterprise constraints apply at the organization level and automatically cascade to all projects within the organization. When you enable a constraint for a product, users cannot create new resources for that product, even if their role would otherwise allow it. Existing resources continue to operate normally when you enable a constraint.
```

What do you think of replacing "inherit" with "cascade"

Author: No problem.

- **Meet security compliance requirements** - Disable products that have not passed internal security reviews
- **Enforce organizational standards** - Maintain clear separation of duties across teams

You can restrict the following HCP products:

- HCP Vault
- HCP Boundary
- HCP Vault Radar
- HCP Packer
- HCP Waypoint

~> **Note:** HCP Terraform is not affected by enterprise constraints.

## Prerequisites

To configure enterprise constraints, you must have one of the following roles:

- Organization Owner
- Organization Admin

You should also understand which HCP products your teams currently use or plan to use.

## Enable enterprise constraints

To enable enterprise constraints:

1. At the top, click the dropdown to open the organization and project selector. Select **View all organizations**.
1. Click the name of the organization.
1. From the **Organization dashboard**, click **Organization settings**.
1. Click **Enterprise constraints** in the left navigation.
1. Select the products you want to restrict from the available list:
   - HCP Vault
   - HCP Boundary
   - HCP Vault Radar
   - HCP Packer
   - HCP Waypoint
1. Click **Save constraints**.

> After saving, the constraints take effect immediately. Users attempting to create resources for restricted products will see an error message indicating that the product is not available in the organization.

Reviewer 2: ```suggestion
After saving, the constraints take effect immediately. Users attempting to create resources for restricted products see an error message indicating that the product is not available in the organization.
```
Style nit: Use present tense

## View active constraints

To view which constraints are currently active:

1. At the top, click the dropdown to open the organization and project selector. Select **View all organizations**.
1. Click the name of the organization.
1. From the **Organization dashboard**, click **Organization settings**.
1. Click **Enterprise constraints** in the left navigation.

The page displays all products and indicates which ones are currently restricted.

## Modify constraints

To modify enterprise constraints:

> 1. Navigate to the **Enterprise constraints** page following the steps above.

Reviewer 2: ```suggestion
1. Navigate to the **Enterprise constraints** page following the steps in the [View active constraints section](#view-active-constraints).
```

[We avoid directional words like above/below](https://github.com/hashicorp/web-unified-docs/blob/main/docs/style-guide/top-12.md#organize-content-so-that-it-flows-in-a-single-direction-from-beginning-to-end).

Enterprise constraints have the following behaviors:

- **Existing resources are not affected**: When you enable a constraint for a product, any existing resources for that product continue to operate normally. The constraint only prevents creation of new resources.
- **All projects inherit constraints**: Organization-level constraints apply to all projects within the organization. You cannot override a constraint for a specific project.
- **Clear error messages**: When a user attempts to create a resource for a restricted product, they receive a clear error message indicating that the product is not available in the organization.
- **Audit logging**: All changes to enterprise constraints are recorded in audit logs, including who made the change and when.

## Use cases

### Control flexible consumption spending

Organizations using HCP's flexible consumption plan can restrict which products teams can deploy to prevent budget overruns. For example, if your organization primarily uses HCP Terraform and HCP Vault, you can restrict HCP Boundary, HCP Packer, and HCP Waypoint to ensure spending stays within expected parameters.

### Meet security compliance requirements

> Large enterprises often require internal security reviews before approving products for use. Enterprise constraints allow you to disable products that have not yet passed your security review process, unblocking adoption of approved products while maintaining compliance standards.

Reviewer 2: Do we need this **Use cases** section? You already mentioned both of these in the Introduction.

Author: Its good to edit down. I was thinking it as showing the value to customers. But I can see its making the doc lengthy. I'll remove.

```hcl
resource "hcp_resource_control_policy" "example" {
  resource_id = hcp_organization.example.id
  
  constraints = [
    {
      name = "constraints/vault.deny-creation"
    },
    {
      name = "constraints/packer.deny-creation"
    }
  ]
}
```

Available constraint names:

- `constraints/vault.deny-creation` - Restricts HCP Vault
- `constraints/boundary.deny-creation` - Restricts HCP Boundary
- `constraints/vault-radar.deny-creation` - Restricts HCP Vault Radar
- `constraints/packer.deny-creation` - Restricts HCP Packer
- `constraints/waypoint.deny-creation` - Restricts HCP Waypoint
-->

## Audit and monitoring

All changes to enterprise constraints are recorded in HCP audit logs. Each audit log entry includes:

- The principal (user or service principal) who made the change
- The organization ID
- The constraint that was enabled or disabled
- The timestamp of the change

To access audit logs, refer to the [HCP audit log documentation](/hcp/docs/hcp/audit-log).

## Considerations

- **Organization-level only**: Constraints can only be applied at the organization level. You cannot create project-specific overrides or exceptions.

> - **Product-level restrictions**: Constraints currently restrict entire products. You cannot restrict specific features.

Reviewer 2: What do you think about moving this to the **Introduction**? I am concerned that readers will miss this section since it is so far down the page.

Author: Good point!

1. Click the name of the organization.
1. From the **Organization dashboard**, click **Organization settings**.
1. Click **Organization constraints** in the left navigation.

The page displays all products and indicates which ones are currently restricted.

## Modify constraints

To modify organization constraints:

1. Navigate to the **Organization constraints** page following the steps above.
1. Select or deselect products to change which ones are restricted.
1. Click **Save constraints**.

Changes take effect immediately. If you remove a constraint, users can immediately begin creating resources for that product again.

## Important behaviors

Organization constraints have the following behaviors:

- **Existing resources are not affected**: When you enable a constraint for a product, any existing resources for that product continue to operate normally. The constraint only prevents creation of new resources.
- **All projects inherit constraints**: Organization-level constraints apply to all projects within the organization. You cannot override a constraint for a specific project.

> - **Clear error messages**: When a user attempts to create a resource for a restricted product, they receive a clear error message indicating that the product is not available in the organization.

Reviewer 3: Just want to make sure I’m aligned on the latest plan. Are we still displaying an error message when a user tries to create a resource? Last I recall, the plan was to remove the 'create' actions from the UI entirely and not show the message.

Author: Correct @ameyers33  . But what about from the API/TFprovider ? I assume the errors would bubble up there. 
Should we update the wording in the docs??

Reviewer 3: Yeah, if we're not displaying the error on HCP, it might be good to call out the API/TFprovider and HCP experience when a user tries to create a resource.

Author: Im confused. So the text we have is fine? Or it should be adjusted? 🤔

Author: Notes for later:

Should we describe that behavior as well?
Trevor Powell  [4:08 PM]
something like, "buttons may not show, etc."? (edited) 
Andy Meyers  [4:10 PM]
Yeah something like that. Just to let them know that create actions will be hidden.
